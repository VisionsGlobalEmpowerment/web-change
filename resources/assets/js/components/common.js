import {animated, Spring} from "react-spring";
import { TimingAnimation, Easing } from 'react-spring/dist/addons.cjs'
import React from "react";
import {movementSpeed} from "./animations";

function getDuration(to, from) {
    const duration = Math.sqrt(Math.pow((to.x - from.x), 2) + Math.pow((to.y - from.y), 2));
    if (duration < 10) {
        return 10 * movementSpeed;
    }

    return duration * movementSpeed;
}

export function LocationEntrance(props) {
    return <g transform={'translate(' + props.x + ',' + props.y + ')'}
              onClick={() => props.locked ? false : props.onClick()}
              className={'location-entrance-' + props.locationKey + (props.locked ? '-locked' : '')}
              fill={"#00bcd4"}>
        <rect x={0} y={0} width={50} height={50} rx={5} />
        <text x={5} y={25} fill={"black"}>{props.name}</text>
    </g>;
}

export function LessonEntrance(props) {
    return <g transform={'translate(' + props.x + ',' + props.y + ')'} onClick={() => props.onClick()}>
        <rect x={0} y={0} width={50} height={50} rx={5} fill={"#00bcd4"}
              className={'location-entrance-' + props.locationKey} />
        <text x={5} y={25} >{props.name} {props.completed && "(c)"}</text>
    </g>;
}

export function Character(props) {
    const {
        isMoving,
        from = {},
        to = {}
    } = props.movement;

    if (!isMoving) {
        return <circle cx={props.currentPosition.x} cy={props.currentPosition.y} r={10} fill={"#383ce5e"} />;
    }

    const duration = getDuration(to, from);

    return <Spring native
                   impl={TimingAnimation}
                   config={{ duration: duration, easing: Easing.linear }}
                   from={{ radius: 10, x: from.x, y: from.y }}
                   to={{ radius: 20, x: to.x, y: to.y }}
                   onRest={() => props.onMovementFinish(to.key)}>
        {({ radius, x, y }) => (
            <animated.circle cx={x} cy={y} r={radius} fill={"#383ce5e"} />
        )}
    </Spring>;
}

export function ModalText(props) {
    return <div className="modal-static">
        <div className="modal" tabIndex="-1" role="dialog">
            <div className="modal-dialog" role="document">
                <div className="modal-content">
                    <div className="modal-header">
                        <h5 className="modal-title">{props.name}</h5>
                        <button type="button" className="close" aria-label="Close" onClick={props.onHide}>
                            <span aria-hidden="true">&times;</span>
                        </button>
                    </div>
                    <div className="modal-body">
                        <p>{props.text}</p>
                    </div>
                    <div className="modal-footer">
                        <button type="button" className="btn btn-secondary" onClick={props.onHide}>Close</button>
                    </div>
                </div>
            </div>
        </div>
    </div>
}
