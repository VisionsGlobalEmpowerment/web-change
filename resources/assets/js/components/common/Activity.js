import {Component} from "react";
import React from "react";

export default class Activity extends Component {
    state = {
        activated: false,
        currentTextIndex: 0,
    };

    activate() {
        if (!this.state.activated) {
            this.props.onFinish();
            this.setState({activated: true});

            this.scheduleNextText();
        }
    }

    currentTextLengthInMs() {
        const currentText = this.props.text[this.state.currentTextIndex];
        return 500 + currentText.length * 100;
    }

    scheduleNextText() {
        setTimeout(() => {
            this.showNextText();
        }, this.currentTextLengthInMs());
    }

    showNextText() {
        const {text} = this.props;

        const nextIndex = this.state.currentTextIndex + 1;

        if (nextIndex >= text.length) {
            this.deactivate();
            return;
        }

        this.setState({currentTextIndex: nextIndex});
        this.scheduleNextText();
    }

    deactivate() {
        this.setState({activated: false, currentTextIndex: 0})
    }

    textX(text) {
        return -(text.length * 4 / 2);
    }

    render() {
        const {
            x,
            y,
            activityKey,
            name,
            text
        } = this.props;

        const label = name + (this.state.activated ? ' (activated)' : '');

        return <g transform={'translate(' + x + ',' + y + ')'} onClick={() => this.activate()}>
            <rect x={0} y={0} width={40} height={60} rx={4} fill={"#bdbdbd"}
                  className={'activity-' + activityKey} />
            <text x={this.textX(label)} y={-15} >{label}</text>
            {this.state.activated &&
            (<g transform={'translate(-100, -200)'}>
                <rect width={250} height={150} fill={"#5d8d9d"}/>
                <foreignObject width={250} height={150} requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility">
                    <p style={{textAlign: "center", verticalAlign: "middle"}} xmlns="http://www.w3.org/1999/xhtml">{text[this.state.currentTextIndex]}</p>
                </foreignObject>
            </g>) }
        </g>;
    }
}
