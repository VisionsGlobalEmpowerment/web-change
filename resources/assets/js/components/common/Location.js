import React from "react";
import {getBackgroundAudio} from "../../model/cache";

export const withBackgroundAudio = (audioFilepath) => (WrappedComponent) => {
    return class extends React.Component {

        componentDidMount() {
            const audio = getBackgroundAudio(audioFilepath);
            return audio.play();
        }

        componentWillUnmount() {
            const audio = getBackgroundAudio(audioFilepath);
            return audio.pause();
        }

        render() {
            return <WrappedComponent {...this.props} />;
        }
    };
}

export const backgroundAlign = Object.freeze({
    bottomLeft: "bottom-left",
    bottomCenter: "bottom-center",
    bottomRight: "bottom-right",
    center: "bottom-center",
});

export const withSvgViewport = (config) => (WrappedComponent) => {
    return class extends React.Component {
        state = { width: 0, height: 0, viewBoxWidth: 0, viewBoxHeight: 0 };

        componentDidMount() {
            this.updateWindowDimensions();
            window.addEventListener('resize', this.updateWindowDimensions.bind(this));
        }

        componentWillUnmount() {
            window.removeEventListener('resize', this.updateWindowDimensions.bind(this));
        }

        updateWindowDimensions() {
            const {
                width = 1920,
                height = 1080
            } = config;

            const originalRatio = width / height;
            const windowRatio = window.innerWidth / window.innerHeight;
            if (originalRatio < windowRatio) {
                this.setState({
                    width: window.innerWidth, height: window.innerHeight,
                    viewBoxWidth: width, viewBoxHeight: height * originalRatio / windowRatio
                });
            } else {
                this.setState({
                    width: window.innerWidth, height: window.innerHeight,
                    viewBoxWidth: width * windowRatio / originalRatio, viewBoxHeight: height
                });
            }
        }

        computeX() {
            const {
                width = 1920,
                align = backgroundAlign.bottomCenter
            } = config;

            switch (align) {
                case backgroundAlign.bottomCenter:
                case backgroundAlign.center:
                    return (width - this.state.viewBoxWidth) / 2;
                case backgroundAlign.bottomRight:
                    return width - this.state.viewBoxWidth;
                case backgroundAlign.bottomLeft:
                default:
                    return 0;
            }
        }

        computeY() {
            const {
                height = 1080,
                align = backgroundAlign.bottomCenter
            } = config;

            switch (align) {
                case backgroundAlign.center:
                    return (height - this.state.viewBoxHeight) / 2;
                case backgroundAlign.bottomRight:
                case backgroundAlign.bottomCenter:
                case backgroundAlign.bottomLeft:
                default:
                    return height - this.state.viewBoxHeight;
            }
        }

        render() {
            const viewBox = {
                x: this.computeX(),
                y: this.computeY(),
                width: this.state.viewBoxWidth,
                height: this.state.viewBoxHeight,
            };

            const viewPort = {
                width: this.state.width,
                height: this.state.height,
            };

            return <WrappedComponent viewBox={viewBox} viewPort={viewPort} {...this.props} />;
        }
    };
}
