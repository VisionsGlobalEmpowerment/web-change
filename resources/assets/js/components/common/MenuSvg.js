import React, {Component} from 'react';
import {exit} from "../../model/menu";
import {openSettings} from "../../model/settings";
import {getData} from "../../model/cache";

const Close = ({x, y}) => {
    return <g transform={`translate(${x} ${y})`} onClick={() => exit()} >
        <circle cx={50} cy={50} r={50} fill={'url(#close)'}/>

    </g>
};

const Settings = ({x, y}) => {
    return <g transform={`translate(${x} ${y})`} onClick={() => openSettings()} >
        <circle cx={50} cy={50} r={50} fill={'url("#settings")'}/>
    </g>
}

export const Menu = ({viewBox, viewPort}) => {
    const scale = viewBox.width / viewPort.width;
    const x = viewBox.x + viewBox.width;
    const y = viewBox.y + Math.round(20 * scale);

    return <g>
        <defs>
            <pattern id="settings" x="0" y="0" width={100} height={100} patternUnits="userSpaceOnUse">
                <image xlinkHref={getData("/raw/img/ui/settings_button_01.png")} x={0} y={0} width={100} height={100} />
            </pattern>

            <pattern id="settings-highlight" x="0" y="0" width={100} height={100} patternUnits="userSpaceOnUse">
                <image xlinkHref={getData("/raw/img/ui/settings_button_02.png")} x={0} y={0} width={100} height={100} />
            </pattern>

            <pattern id="close" x="0" y="0" width={100} height={100} patternUnits="userSpaceOnUse">
                <image xlinkHref={getData("/raw/img/ui/close_button_01.png")} x={0} y={0} width={100} height={100} />
            </pattern>

            <pattern id="close-highlight" x="0" y="0" width={100} height={100} patternUnits="userSpaceOnUse">
                <image xlinkHref={getData("/raw/img/ui/close_button_02.png")} x={0} y={0} width={100} height={100} />
            </pattern>
        </defs>

        <Settings x={x - Math.round(216 * scale)} y={y}/>
        <Close x={x - Math.round(108 * scale)} y={y}/>
    </g>

};
