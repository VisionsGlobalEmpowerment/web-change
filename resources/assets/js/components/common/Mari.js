import React from "react";
import {Component} from "react";
import {Group} from "react-konva";
import {KImage} from "../common";

export default class Mari extends Component {
    state = {
        state: 'default',
    };

    toState (state) {
        this.setState({state: state})
    }

    imageFromState(state) {
        switch (state) {
            case 'default':
                return "/raw/img/ferris-wheel/words/bat_default.png";
            case 'speaking':
                return "/raw/img/ferris-wheel/words/bat_done.png"
        }
    }

    render() {
        const {x, y} = this.props;
        return  <Group x={x} y={y}>
            <KImage image={this.imageFromState(this.state.state)}/>
        </Group>
    }
}
