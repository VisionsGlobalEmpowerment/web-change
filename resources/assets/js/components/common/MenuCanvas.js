import React, {Component} from 'react';
import {exit} from "../../model/menu";
import {Circle, Group, Layer, Path} from "react-konva";
import {KImage} from "../common";

export const Menu = ({viewBox, viewPort}) => {
    const scale = viewBox.width / viewPort.width;
    const x = viewBox.x  * scale + viewBox.width - 108;
    const y = viewBox.y + 20 ;

    return <Group x={x} y={y}
                  onClick={() => exit()}
                  onTap={() => exit()}
    >
        <KImage image={"/raw/img/ui/close_button_01.png"}/>
    </Group>;
};

export const Back = ({viewBox, viewPort, onClick}) => {
    const scale = viewBox.width / viewPort.width;
    const x = viewBox.x  * scale + viewBox.width - 216 ;
    const y = viewBox.y + 20;

    return  <Group x={x} y={y}
                   onClick={onClick}
                   onTap={onClick}
    >
        <KImage image={"/raw/img/ui/back_button_01.png"}/>
    </Group>
};
