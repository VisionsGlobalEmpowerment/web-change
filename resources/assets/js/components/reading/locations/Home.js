import React, {Component} from 'react';
import axios from 'axios';
import Activity from "../../common/Activity";
import {pipe, Vera, withEffect} from "../../common";
import {getData} from "../../../model/cache";

const DoorComponent = (props) => {
    return <g transform={'translate(' + props.x + ',' + props.y + ')'}
              onClick={() => props.onClick()}
              className={'location-entrance-' + props.locationKey}>
        <rect x={0} y={0} width={732} height={810} fillOpacity={1}/>
    </g>;
};

const DoorWithEffect = pipe(
    withEffect("door"),
)(DoorComponent);

const ActivityWithEffect = pipe(
    withEffect("teacher"),
)(Activity);

export default class Home extends Component {
    state = {
        movement: {isMoving: false},
    }

    finishActivity(activity) {
        return axios.post(`/courses/reading/activities/` + activity + '/finish')
            .then(res => {
                const progress = res.data;
                this.props.handleProgress(progress);
            })
    }

    render() {
        const {
            width,
            height,
            activity = {
                key: 'lesson-one-instructions',
                name: 'Lesson one instructions',
                text: [
                    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed mollis pellentesque metus, vel vehicula lectus.',
                    'Etiam sit amet sagittis lorem, quis convallis enim. Mauris ut mi a odio euismod pharetra quis sed est. Maecenas ut nulla nibh.',
                    'Suspendisse risus libero, fringilla ac aliquam non, vestibulum et neque. Etiam tristique condimentum ligula, eu consectetur enim sodales non.',
                    'Pellentesque a mi ac enim gravida blandit. Vestibulum laoreet a nisi sit amet venenatis.',
                    'Sed tincidunt, diam ut malesuada ullamcorper, justo urna fermentum dolor, consequat fringilla mi orci sed nibh.',
                    'Praesent nisl orci, condimentum sit amet nisi et, convallis vulputate nibh. Donec porttitor lectus libero, at lobortis lorem imperdiet non.',
                    'Fusce fringilla, mi nec pellentesque venenatis, libero elit vehicula orci, a pellentesque justo turpis at eros.',
                    'Sed at ex sit amet magna molestie ultricies. Etiam aliquam ante in sapien pulvinar, rutrum varius mi feugiat.',
                    'Fusce condimentum magna eu leo rhoncus, a consectetur nulla ullamcorper. Praesent auctor eros lectus, non volutpat ipsum elementum at.'
                ],
                x: 857,
                y: 177
            }
        } = {width: 2520, height: 1080};

        return (
            <g className={"location-home"}>
                <defs>
                    <pattern id="background" x="0" y="0" width={width} height={height} patternUnits="userSpaceOnUse">
                        <image xlinkHref={getData("/raw/img/casa_background.png")} x={0} y={0} width={width} height={height} />
                    </pattern>

                    <pattern id="door" x={0} y={0} width={732} height={810} patternUnits="userSpaceOnUse">
                        <rect width={732} height={810} fillOpacity={0} />
                    </pattern>

                    <pattern id="door-highlight" x={0} y={0} width={732} height={810} patternUnits="userSpaceOnUse">
                        <image xlinkHref={getData("/raw/img/casa_door.png")} x={0} y={0} width={732} height={810} />
                    </pattern>

                    <pattern id="teacher" x={0} y={0} width={426} height={795} patternUnits="userSpaceOnUse">
                        <image xlinkHref={getData("/raw/img/teacher.png")} x={0} y={0} width={426} height={795} />
                    </pattern>

                    <pattern id="teacher-highlight" x={0} y={0} width={434} height={808} patternUnits="userSpaceOnUse">
                        <image xlinkHref={getData("/raw/img/teacher_two.png")} x={0} y={0} width={434} height={808} />
                    </pattern>
                </defs>

                <rect width={width} height={height} className={"location-background"} />

                <DoorWithEffect
                    x={1446} y={42}
                    locationKey={"map"}
                    name={"Door"}
                    onClick={() => this.props.handleMove("map")} />

                <ActivityWithEffect
                    key={activity.key}
                    x={activity.x} y={activity.y}
                    activityKey={activity.key}
                    name={activity.name}
                    text={activity.text}
                    onFinish={() => this.finishActivity(activity.key)}/>

                <Vera x={1350} y={400} />
            </g>
        )
    }
}
