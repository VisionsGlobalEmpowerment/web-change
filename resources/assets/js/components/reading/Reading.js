import React, { Component } from 'react';
import axios from 'axios';
import ReactDOM from "react-dom";
import Map from "./locations/Map";
import Fair from "./locations/Fair";
import Home from "./locations/Home";
import FerrisWheel, {assets} from "./lessons/FerrisWheel";
import Chapas from "./lessons/Chapas";
import {backgroundAlign, withBackgroundAudio, withSvgViewport} from "../common/Location";
import Preloader, {withPreloader} from "../common/Preloader";
import {pipe} from "../common";
import Settings from "../common/Settings";
import {Score} from "./lessons/ferris-wheel/Score";

export default class Reading extends Component {
    static course = 'reading';

    state = {
        progress: {},
        route: history.state ? history.state : {location: 'home'},
    };

    locations = {
        'home': pipe(
            withBackgroundAudio('/raw/audio/background/POL-daily-special-short.mp3'),
            withSvgViewport({width: 2520, height: 1080}),
        )(Home),
        'fair': pipe(
            withBackgroundAudio('/raw/audio/background/POL-daily-special-short.mp3'),
            withSvgViewport({align: backgroundAlign.bottomRight}),
        )(Fair),
        'map': pipe(
            withBackgroundAudio('/raw/audio/background/POL-daily-special-short.mp3'),
            withSvgViewport({align: backgroundAlign.center}),
        )(Map),
        'ferris-wheel': pipe(
            withBackgroundAudio('/raw/audio/background/POL-daily-special-short.mp3'),
            withSvgViewport({align: backgroundAlign.center}),
            withPreloader(assets, '/raw/img/feria/feria.png'),
        )(FerrisWheel),
        'chapas': Chapas,
    };

    handleProgress = (progress) => {
        this.setState({ progress });
    }

    handleMove = (locationName) => {
        if (this.isAvailable(locationName)) {
            const route = {location: locationName};
            history.pushState(route, locationName, "/reading")
            this.setState({
                route: route
            });
            return true;
        }

        return false;
    }

    isAvailable = (location) => {
        if (this.state.progress.availableLocations == null) {
            return false;
        }

        return this.state.progress.availableLocations.includes(location);
    }

    componentDidMount() {
        window.onpopstate = (event) => {
            if (event.state == null) {
                return;
            }

            this.setState({
                route: event.state
            });
        };

        return axios.get(`/courses/reading/progress`)
            .then(res => {
                const progress = res.data;
                this.setState({ progress });
            })
    }

    render() {
        const {
            location = 'home'
        } = this.state.route ? this.state.route : {};

        const LocationComponent = this.locations[location];
        return (
            <div>
                <LocationComponent progress={this.state.progress} handleMove={this.handleMove} handleProgress={this.handleProgress}/>
                <Settings/>
            </div>
        )
    }
}

const files = [
    {url: '/raw/audio/background/POL-daily-special-short.mp3', size: 10, type: "audio"},
    {url: '/raw/audio/effects/NFF-fruit-collected.mp3', size: 1, type: "audio"},
    {url: '/raw/audio/effects/NFF-glitter.mp3', size: 1, type: "audio"},
    {url: '/raw/audio/effects/NFF-robo-elastic.mp3', size: 1, type: "audio"},
    {url: '/raw/audio/effects/NFF-rusted-thing.mp3', size: 1, type: "audio"},
    {url: '/raw/audio/effects/NFF-zing.mp3', size: 1, type: "audio"},

    {url: '/raw/audio/scripts/intro/teacher.mp3', size: 5, type: "audio"},
    {url: '/raw/audio/scripts/intro/vera.mp3', size: 5, type: "audio"},
    {url: '/raw/audio/scripts/intro/syllables.mp3', size: 2, type: "audio"},

    {url: '/raw/img/map/background.png', size: 10, type: "image"},
    {url: '/raw/img/map/casa_01.png', size: 1, type: "image"},
    {url: '/raw/img/map/casa_02.png', size: 1, type: "image"},
    {url: '/raw/img/map/feria_01.png', size: 1, type: "image"},
    {url: '/raw/img/map/feria_02.png', size: 1, type: "image"},
    {url: '/raw/img/map/feria_03.png', size: 1, type: "image"},
    {url: '/raw/img/map/feria_locked.png', size: 1, type: "image"},

    {url: '/raw/img/casa_background.png', size: 10, type: "image"},
    {url: '/raw/img/casa_door.png', size: 1, type: "image"},
    {url: '/raw/img/chat_bubble_big.png', size: 1, type: "image"},
    {url: '/raw/img/teacher.png', size: 1, type: "image"},
    {url: '/raw/img/teacher_two.png', size: 1, type: "image"},
    {url: '/raw/img/vera.png', size: 1, type: "image"},
    {url: '/raw/img/vera/10_sprite_test.png', size: 4, type: 'image'},

    {url: '/raw/img/feria/background.png', size: 10, type: "image"},
    {url: '/raw/img/feria/back.png', size: 1, type: "image"},
    {url: '/raw/img/feria/back_active.png', size: 1, type: "image"},
    {url: '/raw/img/feria/wheel.png', size: 2, type: "image"},

    {url: '/raw/img/ui/back_button_01.png', size: 1, type: "image"},
    {url: '/raw/img/ui/back_button_02.png', size: 1, type: "image"},
    {url: '/raw/img/ui/close_button_01.png', size: 1, type: "image"},
    {url: '/raw/img/ui/close_button_02.png', size: 1, type: "image"},
    {url: '/raw/img/ui/play_button_01.png', size: 1, type: "image"},
    {url: '/raw/img/ui/play_button_02.png', size: 1, type: "image"},
    {url: '/raw/img/ui/reload_button_01.png', size: 1, type: "image"},
    {url: '/raw/img/ui/reload_button_02.png', size: 1, type: "image"},
    {url: '/raw/img/ui/settings_button_01.png', size: 1, type: "image"},
    {url: '/raw/img/ui/settings_button_02.png', size: 1, type: "image"},
    {url: '/raw/img/ui/star_01.png', size: 1, type: "image"},
    {url: '/raw/img/ui/star_02.png', size: 1, type: "image"},
    {url: '/raw/img/ui/star_03.png', size: 1, type: "image"},

    {url: "/raw/img/ferris-wheel/words/broccoli_default.png", size: 1, type: "image"},
    {url: "/raw/img/ferris-wheel/words/crocodile_default.png", size: 1, type: "image"},
    {url: "/raw/img/ferris-wheel/words/orange_default.png", size: 1, type: "image"},
];

if (document.getElementById('reading-course')) {
    ReactDOM.render(<Preloader files={files}><Reading/></Preloader>, document.getElementById('reading-course'));
}
