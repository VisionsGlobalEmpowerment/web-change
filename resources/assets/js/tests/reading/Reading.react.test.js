import renderer from "react-test-renderer";
import React from "react";
import axios from "axios";
import Reading from "../../components/reading/Reading";
import Home from "../../components/reading/locations/Home";
import {withSvgViewport} from "../../components/common/Location";

jest.mock('axios');
jest.mock('../../components/animations', () => ({
    disabled: true,
}));

jest.mock('../../components/common/Preloader', () => ({
    withPreloader: (files) => (WrappedComponent) => (props) => <WrappedComponent {...props}/>,
}));

jest.mock('../../model/executor', () => {
    return function() {
        return {
            executeSequence: () => Promise.resolve(),
            executeAudio: () => Promise.resolve(),
        };
    };
});

jest.mock('../../model/audio');

function wait(time) {
    return new Promise((fulfilled) => {
        setTimeout (() => fulfilled(), time)
    });
}

beforeEach(() => {
    history.pushState(null, '');
});

test('Reading can be rendered', () => {
    const resp = {data:{availableLocations: ['home', 'map']}};
    axios.get.mockResolvedValue(resp);

    const component = renderer.create(
        <Reading />,
    );
    return Promise
        .resolve(component)
        .then(() => {
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
});

test('Student can navigate to map', (done) => {
    const resp = {data:{availableLocations: ['home', 'map']}};
    axios.get.mockResolvedValue(resp);

    const component = mount(<Reading />);

    return Promise
        .resolve(component)
        .then(() => {
            component.update();
            expect(component.state().route).toEqual({location: 'home'});
            component.find('.location-entrance-map').simulate('click');
            return wait(100);
        }).then(() => {
            expect(component.state().route).toEqual({location: 'map'});
            done();
        });
});

test('Student can open the Ferris Wheel', (done) => {
    axios.get.mockResolvedValueOnce({data:{availableLocations: ['home', 'map', 'fair', 'ferris-wheel']}});
    axios.get.mockResolvedValueOnce({data: []});
    axios.get.mockResolvedValueOnce({data: {}});

    const component = mount(<Reading />);

    return Promise
        .resolve(component)
        .then(() => {
            component.update();
            component.find('.location-entrance-map').simulate('click');
            return wait(100);
        }).then(() => {
            component.update();
            component.find('.location-entrance-fair').simulate('click');
            return wait(100);
        }).then(() => {
            component.update();
            component.find('.location-entrance-ferris-wheel').simulate('click');
            return wait(100);
        }).then(() => {
            expect(component.state().route).toEqual({location: 'ferris-wheel'});
        }).then(() => {
            done();
        });
});


test('Lesson one instruction activity can be finished', () => {
    const resp = {data:{availableLocations: ['home', 'map', 'fair'], finishedActivities: ['lesson-one-instructions']}};
    axios.post.mockResolvedValue(resp);

    const handleProgress = jest.fn();
    const currentProgress = {availableLocations: ['home', 'map']};

    const HomeWithSvg = withSvgViewport({})(Home);
    const component = mount(<HomeWithSvg progress={currentProgress} handleProgress={handleProgress}/>);
    component.find('.senora-vaca').simulate('click');

    setImmediate(() => {
        expect(handleProgress.mock.calls.length).toBe(1);
        expect(handleProgress.mock.calls[0][0]).toEqual(resp.data);
    });
});
