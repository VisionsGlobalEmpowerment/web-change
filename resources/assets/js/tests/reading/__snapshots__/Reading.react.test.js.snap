// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reading can be rendered 1`] = `
<div>
  <svg
    className="location-home"
    viewBox="540 0 1440 1080"
  >
    <defs>
      <pattern
        height={1080}
        id="background"
        patternUnits="userSpaceOnUse"
        width={2520}
        x="0"
        y="0"
      >
        <image
          height={1080}
          width={2520}
          x={0}
          y={0}
        />
      </pattern>
      <pattern
        height={810}
        id="door"
        patternUnits="userSpaceOnUse"
        width={732}
        x={0}
        y={0}
      >
        <rect
          fillOpacity={0}
          height={810}
          width={732}
        />
      </pattern>
      <pattern
        height={810}
        id="door-highlight"
        patternUnits="userSpaceOnUse"
        width={732}
        x={0}
        y={0}
      >
        <image
          height={810}
          width={732}
          x={0}
          y={0}
        />
      </pattern>
      <pattern
        height={795}
        id="teacher"
        patternUnits="userSpaceOnUse"
        width={426}
        x={0}
        y={0}
      >
        <image
          height={795}
          width={426}
          x={0}
          y={0}
        />
      </pattern>
      <pattern
        height={808}
        id="teacher-highlight"
        patternUnits="userSpaceOnUse"
        width={434}
        x={0}
        y={0}
      >
        <image
          height={808}
          width={434}
          x={0}
          y={0}
        />
      </pattern>
    </defs>
    <rect
      className="location-background"
      height={1080}
      width={2520}
    />
    <g
      className="location-entrance-map"
      onClick={[Function]}
      transform="translate(1446,42)"
    >
      <rect
        fillOpacity={1}
        height={810}
        width={732}
        x={0}
        y={0}
      />
    </g>
    <g
      onClick={[Function]}
      transform="translate(857,177)"
    >
      <rect
        className="activity-lesson-one-instructions"
        height={794}
        width={426}
        x={0}
        y={0}
      />
    </g>
    <g>
      <pattern
        height={626}
        id="vera"
        patternUnits="userSpaceOnUse"
        width={431}
        x={0}
        y={0}
      >
        <image
          height={626}
          width={431}
          x={0}
          y={0}
        />
      </pattern>
      <g
        className="vera"
        transform="translate(1350,400) scale(1)"
      >
        <rect
          height={626}
          width={431}
          x={0}
          y={0}
        />
      </g>
    </g>
    <g>
      <defs>
        <pattern
          height={100}
          id="settings"
          patternUnits="userSpaceOnUse"
          width={100}
          x="0"
          y="0"
        >
          <image
            height={100}
            width={100}
            x={0}
            y={0}
          />
        </pattern>
        <pattern
          height={100}
          id="settings-highlight"
          patternUnits="userSpaceOnUse"
          width={100}
          x="0"
          y="0"
        >
          <image
            height={100}
            width={100}
            x={0}
            y={0}
          />
        </pattern>
        <pattern
          height={100}
          id="close"
          patternUnits="userSpaceOnUse"
          width={100}
          x="0"
          y="0"
        >
          <image
            height={100}
            width={100}
            x={0}
            y={0}
          />
        </pattern>
        <pattern
          height={100}
          id="close-highlight"
          patternUnits="userSpaceOnUse"
          width={100}
          x="0"
          y="0"
        >
          <image
            height={100}
            width={100}
            x={0}
            y={0}
          />
        </pattern>
      </defs>
      <g
        onClick={[Function]}
        transform="translate(1676 28)"
      >
        <circle
          cx={50}
          cy={50}
          fill="url(\\"#settings\\")"
          r={50}
        />
      </g>
      <g
        onClick={[Function]}
        transform="translate(1828 28)"
      >
        <circle
          cx={50}
          cy={50}
          fill="url(#close)"
          r={50}
        />
      </g>
    </g>
  </svg>
</div>
`;
